### PR TITLE
Remove 3 compiler warnings

### PR DESCRIPTION
**Pull request by Aliscans**
This PR removes compiler warnings I get with gcc 10.2.0 when compiling Crawl (in l-file.cc, quiver.cc and view.cc). I don't think it has any other effect.

### CODE REVIEW

File: crawl-ref/source/viewgeom.h [hunk 1]
```diff
@@ -22,6 +22,7 @@ class crawl_view_buffer
 public:
     crawl_view_buffer();
     crawl_view_buffer(const coord_def &sz);
+    crawl_view_buffer(crawl_view_buffer const&) = default;
```
Comment (randomMesh) [inline crawl-ref/source/viewgeom.h:25]: I don't like the placement of `const` here. The line above has it the other way round. Be consistent.

**Comment by wheals** [comment]
I ended up doing a different fix for the third commit. Please rebase to remove that from the history and then I'll merge the first two.

**Comment by wheals** [comment]
Thanks! Are you credited in CREDITS.txt yet?

**Comment by Aliscans** [comment]
I am already, thanks.

**Pull request merged**